keps/sig-apps/2804-consolidate-workload-controllers-status/README.md

```diff
@@ -136,16 +150,40 @@ The following table gives an overview on what conditions each of the workload re
 
 ### Notes/Constraints/Caveats (Optional)
 
+#### Progressing condition
 As observed in some issues (https://github.com/kubernetes/website/pull/31226) and talking to the users, there is a misunderstanding about the meaning of the `Progressing` condition. These include:
 - Thinking that the `Progressing` condition reflects the state of the current Deployment instead of the last rollout. Which includes expectation that the `Progressing` condition will keep checking availability of replicas and revert to `progressing`/`failed` state even after the `complete` state is reached. And that the progressing condition will thus also reflect any changes in scaling.
 - Confusion that ProgressDeadlineExceeded does not occur after the Deployment rollout completes when the availability of pods degrades before the  `.spec.progressDeadlineSeconds` times out.
 
 To summarize, the name of the `Progressing` condition doesn't indicate its true meaning that its main responsibility is the indication of rollouts, and it confuses the users.
 
+#### Available condition
+
+The Available condition in Deployment indicates that sufficient number of replicas is `Available` (`Ready` for MinReadySeconds).
+The behaviour depends on the type of the deployment strategy:
+- `Recreate`: The condition becomes `True` when all the replicas are available.
+- `RollingUpdate`: The condition takes into account the rollout constraints `maxUnavailable` and `maxSurge` to achieve availability (`status=True`) even during a rollout.
+The evaluation rule is `availableReplicas >= replicas - maxUnavailable` and `maxSurge` just influences the defaulting of `maxUnavailable`.
+This rule stays the same even after the rollout has finished and as a consequence adds a toleration for certain amount of failing pods in normal operation, while still being considered available.
+
+The semantic meaning of `Available` condition is that the deployment is healthy and operating at sufficient capacity(number of replicas).
+
+Introducing this condition as is, to other workloads is not straightforward:
+- ReplicaSet & ReplicationController: To introduce this in ReplicaSet we would have to mark `status=True` in Available condition when all the replicas are Available since there is no rollout mechanic in ReplicaSets.
```

Comment:
The Available condition only has meaning on types that define availability, ReplicaSet and RC do not define availability, therefore they do not need to be consistent for either Progressing or Available.  They should be consistent on ReplicaFailure.